includes/class-wc-payments.php

```diff
@@ -146,6 +146,22 @@ public static function init() {
 		}
 
 		add_action( 'rest_api_init', [ __CLASS__, 'init_rest_api' ] );
+		add_action( 'woocommerce_before_order_object_save', [ __CLASS__, 'maybe_set_original_order_currency' ] );
+	}
+
+	/**
+	 * Set order's original currency.
+	 *
+	 * @param WC_Order $order The order whose original currency we want to save.
+	 */
+	public static function maybe_set_original_order_currency( $order ) {
+		$currency          = $order->get_currency();
+		$key               = '_wcpay_original_order_currency';
+		$original_currency = $order->get_meta( $key );
+		if ( ! $currency || $original_currency ) {
+			return;
+		}
```

Comment:
Looks like a good place to reuse our own utility method `get_order_original_currency`?

Reply:
@shendy-a8c : the order has `_order_currency` meta already, isn't it the same as we are trying to do here?

Reply:
@kalessil per [my comment here](https://github.com/Automattic/woocommerce-payments/issues/1200#issuecomment-766176096), Multi-currency app can change order's `_order_currency` and that can mess things up, so I save the original order currency in `_wcpay_original_order_currency`.

While we're talking about `_order_currency`, I'm actually confused with the difference between `$order->get_currency()` and `$order->get_meta('_order_currency')`. They seem to return the same value.

Reply:
@vbelolapotkov do you mean change that function block to be like this?
```
	public static function maybe_set_original_order_currency( $order ) {
		$currency          = $order->get_currency();
		$original_currency = WC_Payments_Utils::get_order_original_currency( $order );
		if ( ! $currency || $original_currency ) {
			return;
		}
		$order->update_meta_data( $key, $currency );
	}
```
or
```
	public static function maybe_set_original_order_currency( $order ) {
		$original_currency = WC_Payments_Utils::get_order_original_currency( $order );
		if ( ! $original_currency ) {
			return;
		}
		$order->update_meta_data( $key, $original_currency );
	}
```

If the latter, I prefer not because I like to be conservative and avoiding writing meta data when not necessary. It's safer that way, right?

Reply:
@shendy-a8c : that's pretty strange, to reproduce I need to try refunding a non-usd order. Is that right?

`get_currrency` is a public API, which preferred way to access the currency (underlying data structures and fallback logic can evolve and the app still works).
`get_meta` is a lower-level API, any changes in underlying data structures can break app.

Reply:
> to reproduce I need to try refunding a non-usd order. Is that right?
Yes.
- Install [Multi-currency app](https://woocommerce.com/products/multi-currency/) and set it up, so you buy product with non-USD.
- Purchase something in non-USD, say in EUR.
- Open your order from admin.
- At the very bottom of the order admin page, you will see a dropdown to change the currency of the order. Checkout [my screenshot](https://user-images.githubusercontent.com/73803630/105613628-466fc480-5df6-11eb-970a-bd7ca3d47a0b.png). Change the order's currency to something else, say JPY.
- Refund the order.

Since refunding is not in customer's session, if we only pass `$order->get_currency()` to `wc_price()`, it will use JPY. That's not correct because the order was created in EUR. That's why I need to store 'original order currency' in `_wcpay_original_order_currency`, can't rely on `_order_currency` or `get_currency()`.

Reply:
> If the latter, I prefer not because I like to be conservative and avoiding writing meta data when not necessary. It's safer that way, right?
The latter will call the update_meta_data() even though _wcpay_original_order_currency is already set.

I don't see it as a big problem. It's not a bottleneck to optimize it, but using internal meta key as a string literal bothers me more. How about moving the whole method (it's static after all) to the WC_Payments_Utils and moving internal meta key to private static property on Utils? This would make both usages next to each other and avoid using string literals for meta key.

Reply:
That's a fair concern, @vbelolapotkov. I've implemented your suggestion, ie moving the meta key to a constant in utils and put the set and get method close to each other within utils class, avoiding using string literal.